.github/workflows/maven.yaml

```diff
@@ -68,17 +68,18 @@ jobs:
     strategy:
       matrix:
         profile:
-          - {name: 'unit-tests',    javaver: 11, args: 'verify -PskipQA'}
-          - {name: 'qa-checks',     javaver: 11, args: 'verify javadoc:jar -Psec-bugs,errorprone'}
-          - {name: 'jdk17',         javaver: 17, args: 'verify'}
+          - {name: 'unit-tests',    distribution: adopt,  javaver: 11, args: 'verify -PskipQA'}
+          - {name: 'qa-checks',     distribution: adopt,  javaver: 11, args: 'verify javadoc:jar -Psec-bugs,errorprone'}
+          - {name: 'jdk17',         distribution: adopt,  javaver: 17, args: '-Dmaven.compiler.source=17 -Dmaven.compiler.target=17 verify'}
+          - {name: 'jdk21',         distribution: oracle, javaver: 21, args: '-Dmaven.compiler.source=21 -Dmaven.compiler.target=21 verify'}
```

Comment:
The compiler source args are interesting, wonder if that should be done in the accumulo project.

Reply:
Yeah, I noticed that it was just using the new JDK's to build java 11 compatible code.

Reply:
That is a good find

Reply:
"adopt" in this config just refers to Adoptium's builds, rather than that of the AdoptOpenJDK. So, I'm not sure those articles linked are relevant, as the switch is already accounted for. But, Adoptium doesn't have 21 builds yet, so might as well use Oracle's builds. The best alternative, which I think many people find popular, is the zulu one.

That said, we definitely shouldn't be passing the compiler properties on the command-line. We should be setting all 3 properties, `maven.compiler.source`, `maven.compiler.target`, and `maven.compiler.release` to 11, regardless of which JDK we build with, so those should be in the POM. Really, the only essential one is `maven.compiler.release`, as that replaces the other two for the compiler. But, the reason to still set the other two is that some other Maven plugins (like the formatter) still use those other properties for their own behavior.

Reply:
I'll add `maven.compiler.release` to the pom, it's not in there.

Reply:
Added `maven.compiler.release` to pom in 5b3ccb6.